### ISSUE TITLE
Mobile autocomplete with converter and var shows wrong value

### ISSUE THREAD

**Issue by mashcos** [primefaces/primefaces#1397]
I have a bean with a field of class Person and a converter for this class.

In the facelet i use the following code:

```
<p:autoComplete value="#{taskBean.assignToPerson}"
                            forceSelection="true" var="person" 
                            converter="#{converterBean.personConverter}"
                            itemLabel="#{person.name}" itemValue="#{person}"
                            completeMethod="#{taskBean.completePerson}"
                            minQueryLength="3" />
```

In the normal (desktop) view everything works fine. But in the mobile view (with  renderKitId="PRIMEFACES_MOBILE") the id of the person (as returned by the converter) instead of the itemLabel is shown when taskBean.assignToPerson already has a value set when calling the page. If it is null at the beginning and you input something, then it works correct.

Maybe this is in some way related to #20 because the indication is the same, but in my case it does not depend on the validation result but whether you use the mobile renderkit.

**Comment by sharedloadcom** [comment]
Can confirm this bug. Only appears when reading. Are there any workarounds?

**Comment by mashcos** [comment]
I've seen there is a difference in the encodeInput method of the AutoCompleteRenderer class for mobile an desktop view. In the desktop version it is checked whether var is null or not:

```
        if(var == null) {
           itemLabel = ComponentUtils.getValueToRender(context, ac);
            if(itemLabel != null) {
                writer.writeAttribute("value", itemLabel , null);
            }
        }
        else {
            Map<String,Object> requestMap = context.getExternalContext().getRequestMap();

            if(ac.isValid()) {
                requestMap.put(var, ac.getValue());
                itemLabel = ac.getItemLabel();
            }
            else {
                Object submittedValue = ac.getSubmittedValue();
                itemLabel = (submittedValue == null) ? null : String.valueOf(submittedValue);

                Object value = ac.getValue();

                if(itemLabel == null && value != null) {
                    requestMap.put(var, value);
                    itemLabel = ac.getItemLabel();
                }
            }

            if(itemLabel != null) {
                writer.writeAttribute("value", itemLabel, null);
            }

            requestMap.remove(var);
        }
```

In the mobile version this check is not done, the code is much shorter:
`
        if(valueToRender != null) writer.writeAttribute("value", valueToRender , null);
`

Maybe there is something missing?

**Comment by sharedloadcom** [comment]
The problem ocurs after inital load (wether page reload or ajax update). The user here

http://forum.primefaces.org/viewtopic.php?f=8&t=44357

recommands to check inital load and maybe change the converter class, but I don't know how to do that.

**Comment by vitormc** [comment]
Did you guys found a solution for it?
I'm stuck with the same problem...

**Comment by mashcos** [comment]
Sorry but there still seems to be no solution. I tried to fix it and had some kind of success, but in the end there were still some problems and i've got too little experience in modifiying primeface.

**Comment by tandraschko** [comment]
PrimeFaces Mobile will be deprecated  in 6.2 and removed in 6.3 ( see https://www.primefaces.org/primefaces-6-2-roadmap/)
Therefor i will close this issue as won't fix. If you will provide a PR to fix this issue until 6.2 RC1, i can merge it.
